psi_cardinality/go/server/server.go

```diff
@@ -90,12 +91,12 @@ func CreateWithNewKey() (*PSICardinalityServer, error) {
 //CreateFromKey creates and returns a new server instance with the provided private key.
 //
 //Returns an error if any crypto operations fail.
-func CreateFromKey(key string) (*PSICardinalityServer, error) {
+func CreateFromKey(key []byte) (*PSICardinalityServer, error) {
 	psiServer := new(PSICardinalityServer)
 
 	var err *C.char
 	rcode := C.psi_cardinality_server_create_from_key(C.struct_server_buffer_t{
-		buff:     C.CString(key),
+		buff:     (*C.char)(unsafe.Pointer(&key[0])),
```

Comment:
Might be easier to read like this `(*C.char)(C.CBytes(key))`

Reply:
Fixed.